compiler/src/dotty/tools/dotc/transform/ShortcutImplicits.scala

```diff
@@ -44,8 +44,20 @@ import collection.mutable
  *  (2) A reference `qual.apply` where `qual` has implicit function type and
  *  `qual` refers to a method `m` is rewritten to a reference to `m$direct`,
  *  keeping the same type and value arguments as they are found in `qual`.
+ *
+ *  Note: The phase adds direct methods for all defined methods with IFT results,
+ *        as well as for all methods that are referenced. It does NOT do an
+ *        info transformer that adds these methods everywhere where an IFT returning
+ *        method exists.
+ *        Adding such an info transformer is impractical because it would mean
+ *        that we have to force the types of all members of classes that are referenced.
```

Comment:
I don't quite understand how info transformer works. Maybe a future compiler session on the topic, I know several in the team are interested in the topic.